src/System.Data.SqlClient/src/System/Data/SqlClient/SqlFileStream.Windows.cs

```diff
@@ -0,0 +1,979 @@
+                {
+                    uint ioControlCode = Interop.Kernel32.CTL_CODE(FILE_DEVICE_FILE_SYSTEM,
+                        IoControlCodeFunctionCode, (byte)Interop.Kernel32.Method.METHOD_BUFFERED,
+                        (byte)Interop.Kernel32.Access.FILE_ANY_ACCESS);
+
+                    if (!Interop.Kernel32.DeviceIoControl(hFile, ioControlCode, IntPtr.Zero, 0, IntPtr.Zero, 0, out uint cbBytesReturned, IntPtr.Zero))
+                    {
+                        System.ComponentModel.Win32Exception e = new System.ComponentModel.Win32Exception(Marshal.GetLastWin32Error());
+                        ADP.TraceExceptionAsReturnValue(e);
+                        throw e;
+                    }
+                }
+
+                // now that we've successfully opened a handle on the path and verified that it is a file,
+                // use the SafeFileHandle to initialize our internal System.IO.FileStream instance
+                // NOTE: need to assert UnmanagedCode permissions for this constructor. This is relatively benign
+                // in that we've done much the same validation as in the FileStream(string path, ...) ctor case
+                // most notably, validating that the handle type corresponds to an on-disk file.
+                bool bRevertAssert = false;
+                try
+                {
+                    SecurityPermission sp = new SecurityPermission(SecurityPermissionFlag.UnmanagedCode);
+                    sp.Assert();
+                    bRevertAssert = true;
+
+                    System.Diagnostics.Debug.Assert(_m_fs == null);
+
+                    _m_fs = new System.IO.FileStream(hFile, access, DefaultBufferSize, ((options & System.IO.FileOptions.Asynchronous) != 0));
+                }
+                finally
+                {
+                    if (bRevertAssert)
+                        SecurityPermission.RevertAssert();
+                }
+
+            }
+            catch
+            {
+                if (hFile != null && !hFile.IsInvalid)
+                    hFile.Dispose();
+
+                throw;
+            }
+            finally
+            {
+                if (eaBuffer != null)
+                {
+                    eaBuffer.Dispose();
+                    eaBuffer = null;
+                }
+            }
+        }
+
+        #region private helper methods
+
+        // This method exists to ensure that the requested path name is unique so that SMB/DNS is prevented
+        // from collapsing a file open request to a file handle opened previously. In the SQL FILESTREAM case,
+        // this would likely be a file open in another transaction, so this mechanism ensures isolation.
+        static private string InitializeNtPath(string path)
+        {
+            // ensure we have validated and normalized the path before
+            AssertPathFormat(path);
+
+            string formatPath = @"\??\UNC\{0}\{1}";
```

Comment:
You should check the format of the given path. `PathInternal.IsDeviceUnc()` will see if you're already in the format you're trying to generate here. If so, you just need to switch the prefix to `\??`. (e.g. the path could already be `\\.\UNC\...` or `\\?\UNC\....`)